src/engine/sync/internalclock.cpp

```diff
@@ -184,12 +189,18 @@ void InternalClock::updateBeatLength(int sampleRate, double bpm) {
 void InternalClock::onCallbackStart(int sampleRate, int bufferSize) {
     Q_UNUSED(sampleRate)
     Q_UNUSED(bufferSize)
+    m_bClockUpdated.fetchAndStoreAcquire(false);
```

Comment:
You use acquire for both states true and false. Usually acquire and release memory barriers are used symmetrically. I don't understand how those operations are supposed to work here, or if memory barriers are actually needed? If they are needed the semantics should be applied correctly, otherwise this is highly confusing.

Reply:
I believe the different calls are used depending on how the memory needs to be accessed in the function, see: https://doc.qt.io/qt-5/qatomicinteger.html#memory-ordering.

it doesn't have to be symmetrical between the different calls.  In this case, the code is order-dependent following the store, so we want to make sure the store happens first before the next instructions are computed.  At least that's my understanding of how it works.